app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -213,3 +214,86 @@ export const createBlankApiRecord = (
 
   return upsertApiRecord(uid, newRecord, teamId);
 };
+
+export const extractQueryParams = (inputString: string) => {
+  const queryParams: KeyValuePair[] = [];
+
+  inputString = split(inputString, "?")[1];
+
+  if (inputString) {
+    const queryParamsList = split(inputString, "&");
+    forEach(queryParamsList, (queryParam) => {
+      const queryParamValues = split(queryParam, "=");
+      queryParams.push({
+        id: Math.random(),
+        key: queryParamValues[0],
+        value: queryParamValues[1],
+        isEnabled: true,
+      });
+    });
+  }
+
+  return queryParams;
+};
+
+export const queryParamsToURLString = (queryParams: KeyValuePair[], inputString: string) => {
+  if (isEmpty(queryParams)) {
+    return inputString;
+  }
+  const baseUrl = split(inputString, "?")[0];
+  const enabledParams = queryParams.filter((param) => param.isEnabled);
+
+  const queryString = enabledParams
+    .map(({ key, value }) => {
+      if (!key) return "";
+      if (value === undefined || value === "") {
+        return key;
+      } else {
+        return `${key}=${value}`;
+      }
+    })
+    .filter(Boolean)
+    .join("&");
+
+  return `${baseUrl}${queryString ? `?${queryString}` : queryString}`;
+};
+
+export const syncQueryParams = (queryParams: KeyValuePair[], url: string, type: number = 0) => {
+  const updatedQueryParams = extractQueryParams(url);
+
+  switch (type) {
+    case 0: {
```

Comment:
Along with that, let's have this switch block perform exhaustive checking like [this](https://www.typescriptlang.org/play/?#code/C4TwDgpgBA8pBOBDYBLA9gOygXigcgGN4JkI8oAffYxAE3KrwFcxbSH9aIAbCYMgFACAZkwwFUmKAmFp4AWzgQkkjAAo0YAFywEydBgCUOjE3kAjZVADeAqFADOAdxTACACygawhm3ftQBIgO0IQ0-Hha-gFQAPSx0sqyCoHhENEBxMBM8FgAjBlBIdQk9FEx9vGJ8MnyUDS0GfZZOVgATIXBoSxsEeUVVTJydT2kTfV8rVAAzAIAkFVO0BgQELRQwGhQdOvA7tBF0MmcPHxkUHInwohM3MDk+8QbWw6Iwjwgjogg81V7yFAllB3IgAG4HNDg4jrCBQz6HX4JAAq+0CaHkYBQvHggMweGAagAZI53GhbrQMPjfLxgFAmA5pPA0ARVrssdxAdAQeDgYgMLReOtXAA6fwAXwEEqAA).

Context:
A number is an unbounded set. We can't ever say in a switch statement that we have covered everything simply by looking at the code, neither can the compiler. Meaning there's a risk that we miss handling a case.

For switch statement, where each arm represents a thing that can happen, and the switch consolidates it into all the possible things that can happen, it's better if the thing over which we are switching is finite. 
The switch statement and the compiler can tell us if we missed something, and it's easier to glean over all the possible scenarios simple by looking at the type rather than switch arms.

Reply:
It has been fixed with the recent change in the code now the number has been replaced with a string to handle different switch cases with string it becomes more clear what we want to achieve at the time of calling the function. 
Default case is there to handle all the other cases if there is a typo in string.